src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -912,20 +918,21 @@ ReactDOMComponent.Mixin = {
           deleteListener(this, propKey);
         }
       } else if (isCustomComponent(this._tag, nextProps)) {
-        if (propKey === CHILDREN) {
-          nextProp = null;
+        if (!RESERVED_PROPS.hasOwnProperty(propKey)) {
+          DOMPropertyOperations.setValueForAttribute(
```

Comment:
@gaearon this implementation doesn't allow to pass values other than strings to a custom element. How do I pass an array, object or function?